src/database.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseObserver class to handle database relation and state."""
+
+import typing
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires
+from ops.charm import CharmBase
+from ops.framework import Object
+
+DATABASE_NAME = "discourse"
+
+
+class DatabaseHandler(Object):
+    """The Database relation observer."""
+
+    _RELATION_NAME = "database"
+
+    def __init__(self, charm: CharmBase):
+        """Initialize the observer and register event handlers.
+
+        Args:
+            charm: The parent charm to attach the observer to.
+        """
+        super().__init__(charm, "database-observer")
+        self._charm = charm
+        self.database = DatabaseRequires(
+            self._charm,
+            relation_name=self._RELATION_NAME,
+            database_name=DATABASE_NAME,
+        )
+
+    def get_relation_data(self) -> typing.Dict[str, str]:
```

Comment:
Nitpick:
I'm wondering if we introduce a typed dataclass it would be easier to digest the code :D, I usually code with `tab` keys on my IDE 😂

Reply:
Text editor here, every character is typed !

More seriously, `get_relation_data` is consumed only twice, is it worth it to introduce a dataclass ?

Reply:
I see! Probably not :D